### PR TITLE
Adding index on email/name for forms table

### PR DESCRIPTION
**Pull request by caleybrock**
Testing locally: 
- check that there is no index on name/email on localhost db
  screen shot 2015-11-03 at 1 06 07 pm
- create script
- run 'rake build' in /code-dot-org
- check that new index is created on localhost db
  screen shot 2015-11-03 at 1 01 59 pm

@philbogle @laurelfan @wjordan this should help with this pr: https://github.com/code-dot-org/code-dot-org/pull/4828

### CODE REVIEW

File: pegasus/migrations/78_add_hoc_index.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+Sequel.migration do
+  up do
+    add_index :forms, [:email, :name]
```
Comment (philbogle) [inline pegasus/migrations/78_add_hoc_index.rb]: Didn't want kind in this index as well.

**Comment by philbogle** [comment]
LGTM

**Comment by laurelfan** [comment]
can you also test that your query uses the index?

**Comment by philbogle** [comment]
Caley, you can do that using "*explain [query]" *in the mysql console.  I
can look at this with you if you'd like.

https://dev.mysql.com/doc/refman/5.5/en/using-explain.html

**Comment by caleybrock** [comment]
Yep, it uses the index.

`mysql> explain SELECT count(*) AS`count`FROM (SELECT`name`,`email`,`processed_data`,`data`FROM`forms`WHERE (`kind`= 'HocSignup2015') GROUP BY`name`,`email`) AS`t1`LIMIT 1;
+----+-------------+------------+------+----------------------------------------------+------------------+---------+-------+------+---------------------------------------------------------------------+
| id | select_type | table      | type | possible_keys                                | key              | key_len | ref   | rows | Extra                                                               |
+----+-------------+------------+------+----------------------------------------------+------------------+---------+-------+------+---------------------------------------------------------------------+
|  1 | PRIMARY     | <derived2> | ALL  | NULL                                         | NULL             | NULL    | NULL  |   28 | NULL                                                                |
|  2 | DERIVED     | forms      | ref  | forms_kind_index,forms_kind_email_name_index | forms_kind_index | 152     | const |   28 | Using index condition; Using where; Using temporary; Using filesort |
+----+-------------+------------+------+----------------------------------------------+------------------+---------+-------+------+---------------------------------------------------------------------+
2 rows in set (0.03 sec)`

**Comment by laurelfan** [comment]
Hm, it's interesting that it chose forms_kind_index instead of the new 3 part index. This could just be because you don't have enough data in your localhost db... When this gets to prod maybe run the explain there?

**Comment by caleybrock** [comment]
Sure, I can do that. Are you saying it should have only used `forms_kind_email_name_index` instead of both `forms_kind_email_name_index` and `forms_kind_index`?

**Comment by laurelfan** [comment]
You want to look at the keys column, not the possible_keys column. Possible keys is the keys that it considered using, keys is the key that it actually used.

I am pretty sure it is acting differently because you have much less data and it will use the right index in prod, but it would be worth checking (and while you're there you might as well also run the query to see  how much it helped)

**Comment by caleybrock** [comment]
@laurelfan it looks like it didn't use the index on production either

`mysql> explain SELECT count(*) AS`count`FROM (SELECT`name`,`email`,`processed_data`,`data`FROM`forms`WHERE (`kind`= 'HocSignup2015') GROUP BY`name`,`email`) AS`t1` LIMIT 1;
+----+-------------+------------+------+---------------------------------------------------+------+---------+-------+--------+---------------------------------------------------------------------+
| id | select_type | table      | type | possible_keys                                     | key  | key_len | ref   | rows   | Extra                                                               |
+----+-------------+------------+------+---------------------------------------------------+------+---------+-------+--------+---------------------------------------------------------------------+
|  1 | PRIMARY     | <derived2> | ALL  | NULL                                              | NULL | NULL    | NULL  | 147136 | NULL                                                                |
|  2 | DERIVED     | forms      | ref  | kind,forms_kind_index,forms_kind_email_name_index | kind | 152     | const | 147136 | Using index condition; Using where; Using temporary; Using filesort |
+----+-------------+------------+------+---------------------------------------------------+------+---------+-------+--------+---------------------------------------------------------------------+
2 rows in set (0.00 sec)

mysql> SELECT count(*) AS `count` FROM (SELECT `name`, `email`, `processed_data`, `data` FROM `forms` WHERE (`kind` = 'HocSignup2015') GROUP BY `name`, `email`) AS `t1` LIMIT 1;
+-------+
| count |
+-------+
| 71383 |
+-------+
1 row in set (2.76 sec)`

EDIT: It looks like it doesn't use it because it selects data and processed data. Good news is that now it will only need data/processed data for the country/companies.

**Comment by laurelfan** [comment]
That's interesting... Does it still take 3 seconds to do the query?

**Comment by caleybrock** [comment]
2.76 seconds.

If I don't select data/processed data it uses the index and only takes 1 second, but both of those are needed for the company/country counts.

**Pull request merged**